src/main/java/booking/service/UserServiceImpl.java

```diff
@@ -22,4 +22,12 @@ public void add(User user) {
     public List<User> listUsers() {
         return userDao.listUsers();
     }
+
+    @Override
+    public User get(Long id) {
+        if (userDao.get(id).isEmpty()) {
+            return null;
+        }
+        return userDao.get(id).get();
```

Comment:
You are calling db twice.
let's throw an exception if the user is null and let's do this inline

Reply:
Fixed